src/test/groovy/graphql/LocalContextAwareCascadingTest.groovy

```diff
@@ -0,0 +1,122 @@
+package graphql
+
+import graphql.execution.DataFetcherResult
+import graphql.schema.idl.RuntimeWiring
+import graphql.schema.idl.TypeRuntimeWiring
+import spock.lang.Specification
+
+/**
+ * Tests to make sure that the local context from a root node can be propagated to the children and grandchildren.
+ *
+ * See <a href="https://github.com/graphql-java/graphql-java/issues/1635">#1635</a> for the original bug report.
+ */
+class LocalContextAwareCascadingTest extends Specification {
+
+    // Configuration for the datafetchers for this schema, this will build a local context which will be consumed all the way down to the user
+    // of the comments
+    def graphql = TestUtil.graphQL("""
+                type Query {
+                    blog: Blog!
+                }
+                
+                type Blog {
+                    name: String!
+                    comments: [Comment!]!
+                }
+                
+                type Comment {
+                    author: User!
+                    title: String!
+                    body: String!
+                    
+                }
+                
+                type User {
+                    username: String!
+                }
+            """,
+            RuntimeWiring.newRuntimeWiring()
+                    .type(TypeRuntimeWiring.newTypeWiring("Query")
+                            .dataFetcher("blog", {
+                                def blog = new Blog()
+                                blog.name = "some name"
+                                DataFetcherResult.newResult()
+                                        .data(blog)
+                                        .localContext("BLOG CONTEXT") // here is where we build the initial context
+                                        .build()
+                            })
+                    )
+                    .type(TypeRuntimeWiring.newTypeWiring("Blog")
+                            .dataFetcher("comments", {
+                                def comment = new Comment()
+                                comment.title = it.getLocalContext() + " (comments data fetcher)"
+                                DataFetcherResult.newResult()
+                                        .data([comment])
+                                        .build()
+                            })
```

Comment:
This is demonstraing what I expect.  This `comments` fetch sets no local context so later the user fetcher inherits the local context from above (aka Blog context)